test/tests/userobjects/nearest_point_layered_side_integral_functor/nearest_point_layered_side_integral_functor.i

```diff
@@ -0,0 +1,79 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 3
+  nx = 40
+  ny = 10
+  nz = 10
+  allow_renumbering = false
+[]
+
+[Materials]
+  [u_mat]
+    type = GenericFunctorMaterial
+    prop_names = 'u'
+    prop_values = 'u_fn'
+  []
+[]
+
+[AuxVariables]
+  [u_layered_integral]
+    order = CONSTANT
+    family = MONOMIAL
+  []
+[]
+
+[AuxKernels]
+  [u_layered_integral_kern]
+    type = SpatialUserObjectAux
+    variable = u_layered_integral
+    user_object = nplaf
+    boundary = 'bottom top'
+    execute_on = 'INITIAL'
+  []
+[]
+
+[Functions]
+  [u_fn]
+    type = ParsedFunction
+    expression = 'x + y + z'
+  []
+[]
+
+[UserObjects]
+  [nplaf]
+    type = NearestPointLayeredSideIntegralFunctor
+    direction = x
+    points='
+      0.25 0 0.25
+      0.75 0 0.25
+      0.25 0 0.75
+      0.75 0 0.75'
+    num_layers = 10
```

Comment:
add a note on how this carefully avoid placing a layer edge on a centroid (and if it does not then it will need to change)